### ISSUE TITLE
Compatibility with macOS Sierra

### ISSUE THREAD

**Issue by lifepillar** [OpenEmulatorProject/OpenEmulator-OSX#12]
Currently, OpenEmulator does not build on macOS 10.12. Is a fix in the works?

**Comment by tobier** [comment]
I've not yet upgraded to 10.12, so as of today there is no fix in the works.

For others who may have upgraded, can you be more specific on what the issues are so that we can make sure that it's not just your issue.

**Comment by lifepillar** [comment]
I am using the formula from https://github.com/lifepillar/homebrew-appleii to build the current master:

```
brew install -vd openemulator --HEAD
```

This is the command used:

```
xcodebuild -sdk macosx10.12 -configuration Release -alltargets MACOSX_DEPLOYMENT_TARGET=10.12 SYMROOT=build PREFIX=/usr/local/Cellar/openemulator/HEAD-9b3912b ARCHS=x86_64 HEADER_SEARCH_PATHS=/usr/include/libxml2 /usr/local/Cellar/libpng/1.6.25/include/libpng16 /usr/local/Cellar/libsamplerate/0.1.8/include /usr/local/Cellar/libsndfile/1.0.26/include /usr/local/Cellar/libzip/1.1.2/include /usr/local/Cellar/libzip/1.1.2/lib/libzip/include /usr/local/Cellar/portaudio/19.20140130/include LIBRARY_SEARCH_PATHS=/usr/local/lib
```

Build settings from command line:

```
    ARCHS = x86_64
    HEADER_SEARCH_PATHS = /usr/include/libxml2 /usr/local/Cellar/libpng/1.6.25/include/libpng16 /usr/local/Cellar/libsamplerate/0.1.8/include /usr/local/Cellar/libsndfile/1.0.26/include /usr/local/Cellar/libzip/1.1.2/include /usr/local/Cellar/libzip/1.1.2/lib/libzip/include /usr/local/Cellar/portaudio/19.20140130/include
    LIBRARY_SEARCH_PATHS = /usr/local/lib
    MACOSX_DEPLOYMENT_TARGET = 10.12
    PREFIX = /usr/local/Cellar/openemulator/HEAD-9b3912b
    SDKROOT = macosx10.12
    SYMROOT = build
```

The command fails with at this point:

```
CompileC build/OpenEmulator.build/Release/OpenEmulator.build/Objects-normal/x86_64/EmulationOutlineView.o src/macosx/EmulationOutlineView.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
    cd /tmp/openemulator-20161017-2187-9sv4kq
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu99 -Wno-trigraphs -fpascal-strings -Os -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-implicit-atomic-properties -Wno-arc-repeated-use-of-weak -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wno-deprecated-implementations -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -mmacosx-version-min=10.12 -g -fvisibility=hidden -Wno-sign-conversion -Wno-infinite-recursion -I/tmp/openemulator-20161017-2187-9sv4kq/build/OpenEmulator.build/Release/OpenEmulator.build/OpenEmulator.hmap -I/tmp/openemulator-20161017-2187-9sv4kq/build/Release/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/libxml2 -I/usr/local/Cellar/libpng/1.6.25/include/libpng16 -I/usr/local/Cellar/libsamplerate/0.1.8/include -I/usr/local/Cellar/libsndfile/1.0.26/include -I/usr/local/Cellar/libzip/1.1.2/include -I/usr/local/Cellar/libzip/1.1.2/lib/libzip/include -I/usr/local/Cellar/portaudio/19.20140130/include -I/tmp/openemulator-20161017-2187-9sv4kq/build/OpenEmulator.build/Release/OpenEmulator.build/DerivedSources/x86_64 -I/tmp/openemulator-20161017-2187-9sv4kq/build/OpenEmulator.build/Release/OpenEmulator.build/DerivedSources -F/tmp/openemulator-20161017-2187-9sv4kq/build/Release -F/tmp/openemulator-20161017-2187-9sv4kq/lib -F/opt/local/Library/Frameworks -F/opt/local/lib -F/usr/local/lib -F/Users/raw/devel/OpenEmulator/src/macosx -MMD -MT dependencies -MF /tmp/openemulator-20161017-2187-9sv4kq/build/OpenEmulator.build/Release/OpenEmulator.build/Objects-normal/x86_64/EmulationOutlineView.d --serialize-diagnostics /tmp/openemulator-20161017-2187-9sv4kq/build/OpenEmulator.build/Release/OpenEmulator.build/Objects-normal/x86_64/EmulationOutlineView.dia -c /tmp/openemulator-20161017-2187-9sv4kq/src/macosx/EmulationOutlineView.m -o /tmp/openemulator-20161017-2187-9sv4kq/build/OpenEmulator.build/Release/OpenEmulator.build/Objects-normal/x86_64/EmulationOutlineView.o
/tmp/openemulator-20161017-2187-9sv4kq/src/macosx/EmulationOutlineView.m:37:46: warning: instance method '-delete:' not found (return type defaults to 'id') [-Wobjc-method-access]
    return [[[self window] windowController] delete:sender];
                                             ^~~~~~
In file included from /tmp/openemulator-20161017-2187-9sv4kq/src/macosx/EmulationOutlineView.m:11:
In file included from /tmp/openemulator-20161017-2187-9sv4kq/src/macosx/EmulationOutlineView.h:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/System/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/System/Library/Frameworks/AppKit.framework/Headers/AppKit.h:127:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSWindowController.h:16:12: note: receiver is instance of class declared here
@interface NSWindowController : NSResponder <NSCoding, NSSeguePerforming> {
           ^
/tmp/openemulator-20161017-2187-9sv4kq/src/macosx/EmulationOutlineView.m:37:5: error: void method 'delete:' should not return a value [-Wreturn-type]
    return [[[self window] windowController] delete:sender];
    ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

There are also some (seemingly not fatal) deprecation warnings.

**Comment by pudquick** [comment]
Until all the warnings are fixed, you'll need to use `GCC_TREAT_WARNINGS_AS_ERRORS = NO`

Otherwise all warnings will be treated as blocking errors.

Could probably be added in as an arg here:
https://github.com/lifepillar/homebrew-appleii/blob/master/openemulator.rb#L32-L43

**Comment by boredzo** [comment]
Yes, you can pass that in as an argument (same as it already is doing with `MACOSX_DEPLOYMENT_TARGET` and others).

Better than turning off “Treat Warnings as Errors”, though, is to fix the warnings. They're all aspects of one problem with one solution:

1. `-[EmulationOutlineView delete:]` returns `IBAction`, which is a synonym for `void`
2. That method sends `delete:` to an NSWindowController and returns the result
3. Neither NSWindowController nor any superclass of it declares `delete:`, so the compiler assumes it returns `id`
4. You can't return `id` from a method returning `void`.

The solution here is to have a category on NSResponder that declares the `delete:` method as an `IBAction`. Then the method will be known, the `return` statement will be valid, and there will be no warnings left in that method.

**Comment by pudquick** [comment]
Absolutely agree that fixing the project is the best for all ("Until all the warnings are fixed") 👍👍

**Comment by tobier** [comment]
Fixed by #13

**Comment by lifepillar** [comment]
Thanks for fixing it! I confirm that the current master builds fine now, but OpenEmulator crashes when I choose Apple ii-j or Apple-ii plus from the Template Chooser (the rest seems to work). I am using the ROM files suggested here: https://github.com/OpenEmulatorProject/libemulation/issues/15.
